src/Cassandra/ProtocolEvents/ProtocolEventDebouncer.cs

```diff
@@ -218,14 +215,20 @@ private static async Task ProcessQueue(EventQueue queue)
                     await queue.MainEvent.Handler().ConfigureAwait(false);
                     foreach (var cb in queue.Callbacks)
                     {
-                        cb?.TrySetResult(true);
+                        if (cb != null)
+                        {
+                            Task.Run(() => cb.TrySetResult(true)).Forget();
```

Comment:
In the future, it could be interesting to create a `TaskCompletionSource` that sets results / exceptions in a new thread, somethings like `TaskCompletionAsyncSource` :)

Reply:
That's a good idea, it's probably something that I should have done by now 😅 I'll definitely do that in the near future